crates/fulf/src/renamemeplsmynamesucks.rs

```diff
@@ -0,0 +1,342 @@
+
+    let threadcount = r.bonus_threads + 1;
+    let mut files_chunks = vec![Vec::with_capacity(1024); threadcount as usize];
+
+    let usize_tc = threadcount as usize;
+    let mut index = 0;
+    let mut errcount = 0_u32;
+    iter.for_each(|res| match res {
+        Ok(dir_entry) => {
+            let path = dir_entry.into_path();
+
+            if path.is_file() {
+                index += 1;
+                if index == usize_tc {
+                    index = 0;
+                }
+
+                files_chunks[index].push(path);
+            }
+        }
+        Err(_) => {
+            errcount += 1;
+            if errcount > 16000 {
+                panic!()
+            }
+        }
+    });
+
+    let files_chunks = files_chunks.into_iter().map(|vec_with_path| {
+        vec_with_path
+            .into_iter()
+            .filter_map(|path| match fs::read(path) {
+                Ok(buffer) => Some(buffer),
+                _ => None,
+            })
+    });
+
+    let capnum = r.results_cap;
+
+    match (r.force_utf8, needle.is_ascii()) {
+        // ascii
+        (false, true) => {
+            let match_and_score = match_and_score!(ascii, needle, max_line_len);
+
+            spawner(files_chunks, capnum, match_and_score, sort_and_print)
+        }
+
+        // utf8
+        (true, _) | (_, false) => {
+            let match_and_score = match_and_score!(utf8, needle, max_line_len);
+
+            spawner(files_chunks, capnum, match_and_score, sort_and_print)
+        }
+    }
+}
+
+#[inline]
+fn spawner<T: Send + 'static>(
+    files_chunks: impl Iterator<Item = impl Iterator<Item = impl AsRef<[u8]>> + Send + 'static>,
+    capnum: usize,
+    match_and_score: impl Fn(&[u8]) -> Option<T> + Send + 'static + Clone,
+    mut sort_and_print: impl FnMut(&mut [T], usize),
+) -> (Vec<T>, usize) {
+    let (sx, rx) = flume::unbounded();
+    let mut threads = Vec::with_capacity(10);
+
+    files_chunks.for_each(|files| {
+        let t;
+        let sender = sx.clone();
+        let match_and_score = match_and_score.clone();
+        t = thread::spawn(move || threadworks::spawn_me(files, sender, capnum, match_and_score));
+
+        threads.push(t);
+    });
+    drop(sx);
+
+    let mut shared = Vec::with_capacity(capnum * 2);
+    let mut total = 0_usize;
+
+    while let Ok(mut inner) = rx.recv() {
+        // append_to_shared(&mut shared, &mut inner, &mut total, capnum, sort_and_print);
+        if !inner.is_empty() {
+            let inner_len = inner.len();
+
+            total = total.wrapping_add(inner_len);
+
+            shared.append(&mut inner);
+            sort_and_print(&mut shared, total.clone());
+            shared.truncate(capnum);
+        }
+    }
+
+    threads.into_iter().for_each(|t| t.join().unwrap());
+
+    (shared, total)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::time::{Duration, SystemTime};
+
+    #[test]
+    fn basic_functionality_test() {
```

Comment:
Okay, I reworked that thing, now it's a bit different, and should work, because this basic test works.
I tested dynamic results on Rust src, and it works too.
Now I just need you to test the usability and speed of this new thing.

A bit or renaming will be done later, and there's no docs right now, but it works and could be tested.
Look in the test to get the idea of what `sort_and_print` function should look like and what to pass into (poorly named) `very_simple` function.

Reply:
I have to admit that I have not understand this new lib, which part it can replace and how it can be used in current flow. I'll spend more time after I merge your dyn filter feature into clap.

Reply:
> 
> 
> I have to admit that I have not understand this new lib, which part it can replace and how it can be used in current flow. I'll spend more time after I merge your dyn filter feature into clap.

Welp, everything, I guess. It uses the same recursive directory searcher as fd/rg, and when it finds a file, it reads the file, then iterates over lines of that file and applies matcher and scorer onto that line. When file ends, it searches for the next file.
With the last commit I changed `setter` function, so now it can take any algorithm.